src/apps/store/services/promotion.service.ts

```diff
@@ -53,5 +57,30 @@ export class PromotionService {
 
     const randomString = randomArray.join('');
     return randomString;
-  };
+  }
+
+  async checkCouponCode(
+    couponCode: string,
+  ): Promise<ResourceDataOutput<CheckPromotionOutput> | boolean> {
+    const promotion = await this.promotionRepository.findOne({
+      couponCode: couponCode,
+    });
+
+    if (!promotion) {
+      return false; // Discount code does not exist
+    }
+
+    const currentDate = new Date();
+    const expirationDate = promotion.expirationDate;
+
+    if (currentDate > expirationDate) {
+      return false; // Expired discount code
+    }
```

Comment:
It's better to combine this condition into the db query, if have no throw the exception.
 ```
 const promotion = await this.promotionRepository.findOne({
      couponCode: couponCode,
      expirationDate: ....
    });
 ```